substrate/frame/support/src/storage/storage_noop_guard.rs

```diff
@@ -37,11 +37,40 @@
 /// });
 /// ```
 #[must_use]
-pub struct StorageNoopGuard(sp_std::vec::Vec<u8>);
+pub struct StorageNoopGuard {
+	storage_root: sp_std::vec::Vec<u8>,
+	error_message: &'static str,
+}
 
 impl Default for StorageNoopGuard {
 	fn default() -> Self {
-		Self(sp_io::storage::root(sp_runtime::StateVersion::V1))
+		Self {
+			storage_root: sp_io::storage::root(sp_runtime::StateVersion::V1),
+			error_message: "`StorageNoopGuard` detected an attempted storage change.",
+		}
+	}
+}
+
+impl From<&'static str> for StorageNoopGuard {
+	fn from(error_message: &'static str) -> Self {
```

Comment:
Not sure about this, i think the `from_error_message` is enough.  
The `From` is more used in cases where the from type can be *converted*. This is not really a conversion but rather a construction, just to recap how it would look like:  
```rust
let _g: StorageNoopGuard = "Wrong changes".into();
```
is slightly weird to me.

Reply:
Yeah you're totally right @ggwpez. Good lesson in thinking about tools for 'constructing' vs 'converting'. Thanks 👌

Reply:
Removed in 7bc5b6157c